pilot/proxy/envoy/discovery_test.go

```diff
@@ -496,6 +496,16 @@ func TestRouteDiscoveryRedirect(t *testing.T) {
 	}
 }
 
+func TestRouteToEgressDiscoveryRedirect(t *testing.T) {
```

Comment:
TestRouteDiscoveryEgressRedirect.. Or better still, fold this test case into the function above (TestRouteDiscoveryRedirect), where you add another egressRule and egressRedirectConfig